pkg/network/telemetry/reporters.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-go/v5/statsd"
+)
+
+// common prefix used across all statsd metric
+const statsdPrefix = "datadog.system_probe.network_tracer."
+
+var statsdDelta deltaCalculator
+
+// ReportStatsd flushes all metrics tagged with `ReportStatsd`
+func ReportStatsd() {
+	client := getClient()
+	if client == nil {
+		return
+	}
+
+	metrics := GetMetrics(OptStatsd)
+	previousValues := statsdDelta.GetState("")
+	for _, metric := range metrics {
+		v := previousValues.ValueFor(metric)
+		if contains(OptGauge, metric.tags) {
+			client.Gauge(statsdPrefix+metric.name, float64(v), metric.tags, 1.0) //nolint:errcheck
+		}
+
+		client.Count(statsdPrefix+metric.name, v, metric.tags, 1.0) //nolint:errcheck
+	}
+}
+
+var telemetryDelta deltaCalculator
+
+// ReportPayloadTelemetry returns a map with all metrics tagged with `OptTelemetry`
+// The return format is consistent with what we use in the protobuf messages sent to the backend
+func ReportPayloadTelemetry(clientID string) map[string]int64 {
+	metrics := GetMetrics(OptTelemetry)
+	previousValues := telemetryDelta.GetState(clientID)
+	result := make(map[string]int64, len(metrics))
+	for _, metric := range metrics {
+		result[metric.name] = previousValues.ValueFor(metric)
+	}
+	return result
+}
+
+var expvarDelta deltaCalculator
+
+// ReportExpvar returns a nested map structure with all metrics tagged with `OptExpvar`
+func ReportExpvar() map[string]interface{} {
+	metrics := GetMetrics(OptExpvar)
+	previousValues := expvarDelta.GetState("")
+	root := make(map[string]interface{})
+	countByName := make(map[string]int)
```

Comment:
seems like there is no meaning for the count (only for the existence of previous metric with the same name), instead I'd use `map[string]struct{}` to represent a set

Reply:
Makes total sense :+1: